src/controllers/projectController.ts

```diff
@@ -31,10 +32,16 @@ export class ProjectController implements Disposable {
             detail: "A project without any build tools",
         }];
         if (contextManager.getContextValue(Context.MAVEN_ENABLED)) {
-            projectKinds.push({
+            const isMavenDefault: boolean = await expManager.isFlightEnabled("defaultMaven");
```

Comment:
Are you OK paying the async cost here? Given that this could result in an http request, this could be 1-5 seconds.